Src/coffeescript/cql-execution/src/elm/nullological.coffee

```diff
@@ -20,6 +20,12 @@ module.exports.Coalesce = class Coalesce extends Expression
 
   exec: (ctx) ->
     for arg in @args
-      result = arg.execute(ctx)
-      if result? then return result
+      # Lists are treated differently
```

Comment:
I assume this is to cover the last signature in the documented cases:
```
Coalesce<T>(argument1 T, argument2 T) T
Coalesce<T>(argument1 T, argument2 T, argument3 T) T
Coalesce<T>(argument1 T, argument2 T, argument3 T, argument4 T) T Coalesce<T>(argument1 T, argument2 T, argument3 T, argument4 T, argument5 T) T
Coalesce<T>(arguments List<T>) T
```
In this case, shouldn't we also be ensuring that length of `@args` is `1`?  If the length of `@args` is more than one, we should probably be treating it like the multi-variable case instead.

Reply:
Additionally, if there is only a single argument, but it is an `ExpressionRef` pointing to a list (rather than a List literal), then I think it wouldn't have `arg.elements` -- you'd have to execute the arg to find out it's a list.  I think.